src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -237,14 +237,20 @@ void M2(__arglist)
         [Fact]
         public void NullCheckedMethodValidationWithOptionalNullParameter()
         {
-            // PROTOTYPE : Make it an error for the default parameter to be set to null with a null-checked value.
             var source = @"
 class C
 {
     void M(string name! = null) { }
```

Comment:
>string name! = null [](start = 11, length = 19)

Please test:
```
static void M1<T>(T t! = default) { }
static void M2<T>(T t! = default) where T : class { }
static void M3<T>(T? t! = default) where T : struct { }
```

Reply:
Please also test the following. I think it's ok if we don't report `WRN_NullCheckedHasDefaultNull` for `B2<T>.F` or `B3<T>.F` but it would be good to ensure we've tested these cases:
```C#
abstract class A<T>
{
    internal abstract void F<U>(U u) where U : T;
}
class B1<T> : A<T>
{
    internal override void F<U>(U u! = default) { }
}
class B2<T> : A<T> where T : class
{
    internal override void F<U>(U u! = default) { }
}
class B3<T> : A<T?> where T : struct
{
    internal override void F<U>(U u! = default) { }
}
```

---
In reply to: [302649197](https://github.com/dotnet/roslyn/pull/36979#discussion_r302649197) [](ancestors = 302649197)